File: example/lib/models/order/order.dart
import 'package:schemix/schemix.dart';

import '../user/user.dart';

part 'order.schemix.dart';

@Schemix(
  name: 'orders',
  schemaVersion: 1,
)
class Order {
  @PrimaryKey(autoGenerate: true)
  final String id;

  @BelongsTo(User)
  final String userId;

  @DatabaseDefault('pending')
  final String status;

  @ReadOnlyField()
  @DatabaseGenerated(strategy: 'sequence')
  final int? orderNumber;

  @WriteOnlyField()
  @Encrypted()
  final String? creditCardToken;

  const Order({
    required this.id,
    required this.userId,
    this.status = 'pending',
    this.orderNumber,
    this.creditCardToken,
  });

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
  Map<String, dynamic> toJson() => _$OrderToJson(this);
  Order copy() => _$OrderCopy(this);
}

File: examples/infolder/backend/go/models/order.go
package models

type Order struct {
	Id string `gorm:"primaryKey;column:id" json:"id"`
	UserId string `gorm:"column:user_id" json:"userId"`
	Status string `gorm:"column:status;not null;default:pending" json:"status"`
	OrderNumber *int64 `gorm:"column:order_number" json:"orderNumber"`
	CreditCardToken *string `gorm:"column:credit_card_token" json:"creditCardToken"`
}

